airflow/models/baseoperator.py

```diff
@@ -381,7 +378,7 @@ def __init__(
             self.retry_delay = retry_delay
         else:
             self.log.debug("Retry_delay isn't timedelta object, assuming secs")
-            self.retry_delay = timedelta(seconds=retry_delay)
+            self.retry_delay = timedelta(seconds=retry_delay.seconds)
```

Comment:
This will raise error the following error!:

```python
TypeError: unsupported type for timedelta seconds component: datetime.timedelta
```
https://github.com/apache/airflow/pull/6601/files#diff-b5ec97b8301ab688b7e045426f24d485R377 checks if it is **timedelta** object or not. If it is not the code assumes retry_delay might be a float or int.

Reply:
I will take a look!

Reply:
Indeed. I have to read the comments.